markusapi/markusapi.py

```diff
@@ -238,6 +238,54 @@ def update_marking_state(self, assignment_id: int, group_id: int, new_marking_st
         path = Markus.get_path(assignments=assignment_id, groups=group_id, update_marking_state=None)
         return self.submit_request(params, path, 'PUT')
 
+    def upload_file_to_repo(self, assignment_id: int, group_id: int, file_path: str, 
+                            contents: Union[str, bytes], mime_type: Optional[str] = None) -> List[str]:
+        """ 
+        Upload a file at file_path with content contents to the assignment directory 
+        in the repo for group with id group_id. 
+
+        The file_path should be a relative path from the assignment directory of a repository.
+        For example, if you want to upload a file to A1/somesubdir/myfile.txt then the short identifier
+        of the assignment with id assignment_id should be A1 and the file_path argument should be:
+        'somesubdir/myfile.txt'
+        """
+        path = Markus.get_path(assignments=assignment_id, groups=group_id, submission_files=None)
+    
+        if mime_type is None:
+            mime_type = mimetypes.guess_type(file_path)[0]
+        
+        if mime_type is None:
+            raise ValueError(f'if the mime_type argument is not given you must provide a title file with a valid extension')
```

Comment:
This is copied over from the feedback file method, and it would be good to make another commit that changes both of these in the following ways:

1. Don't need the `f` string here.
2. I'm not sure what's meant by "title file" but you can change it to "filename" or "`file_path` argument".